### PR TITLE
gh-89038: [doc] update dis.findlinestarts documentation for changes related to PEP-626

### LINKED ISSUE #89038: Update dis.findlinestarts documentaiton to reflect new usage of `co_lines` (PEP 626)
BPO | [44875](https://bugs.python.org/issue44875)
--- | :---
Nosy | @markshannon, @gwk, @laike9m

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-08-09.22:33:48.395>
labels = ['3.10', 'docs']
title = 'Update dis.findlinestarts documentaiton to reflect new usage of `co_lines` (PEP 626)'
updated_at = <Date 2022-01-04.05:24:34.547>
user = 'https://github.com/gwk'
```

bugs.python.org fields:
```python
activity = <Date 2022-01-04.05:24:34.547>
actor = 'laike9m'
assignee = 'docs@python'
closed = False
closed_date = None
closer = None
components = ['Documentation']
creation = <Date 2021-08-09.22:33:48.395>
creator = 'gwk'
dependencies = []
files = []
hgrepos = []
issue_num = 44875
keywords = []
message_count = 3.0
messages = ['399290', '399291', '409647']
nosy_count = 4.0
nosy_names = ['docs@python', 'Mark.Shannon', 'gwk', 'laike9m']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue44875'
versions = ['Python 3.10']
```

</p></details>

### PR DESCRIPTION
**Pull request by iritkatriel**
Closes #89038.

* Issue: gh-89038

### CODE REVIEW

**Comment by iritkatriel** [comment]
@markshannon Does the new wording look ok to you?

File: Doc/library/dis.rst [hunk 1]
```diff
@@ -266,15 +266,19 @@ operation is being performed, so the intermediate analysis object isn't useful:
...
 .. function:: findlinestarts(code)
 
-   This generator function uses the ``co_firstlineno`` and ``co_lnotab``
-   attributes of the code object *code* to find the offsets which are starts of
+   This generator function uses the ``co_lines`` method
+   of the code object *code* to find the offsets which are starts of
    lines in the source code.  They are generated as ``(offset, lineno)`` pairs.
-   See :source:`Objects/lnotab_notes.txt` for the ``co_lnotab`` format and
-   how to decode it.
+   See :source:`Objects/lnotab_notes.txt` for details about the ``co_lines``
```
Comment (markshannon) [inline Doc/library/dis.rst]: Could you link to `Objects.locations.md` instead, for the full details of the format?

Or, just link to the docs for `co_lines` and save the user from the painful details.

Comment (iritkatriel) [reply Doc/library/dis.rst]: I can't seem to find any docs for co_lines.

Comment (iritkatriel) [reply Doc/library/dis.rst]: Are you sure Objects/locations is the right file to reference? It doesn't mention co_lines. Objects/lnotab_notes.txt does.

Comment (markshannon) [reply Doc/library/dis.rst]: They both describe the format of the data, but `Objects.locations.md` is up to date.
I don't think either are really relevant here. You don't need to understand the table format to use `dis`.
Maybe just drop this line?

File: Doc/library/dis.rst [hunk 2]
```diff
@@ -266,15 +266,19 @@ operation is being performed, so the intermediate analysis object isn't useful:
...
+   method.
```
Comment (iritkatriel) [inline Doc/library/dis.rst]:
```suggestion
```

**Review by markshannon — APPROVED** [review]

**Pull request merged**